driver-core/src/main/com/mongodb/client/model/search/RangeSearchOperator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Immutable;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonValue;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.types.Decimal128;
+
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+@Immutable
+public final class RangeSearchOperator extends SearchOperator {
+
+    private final List<SearchPath> path;
+    @Nullable private final BsonValue gt;
+    @Nullable private final BsonValue gte;
+    @Nullable private final BsonValue lt;
+    @Nullable private final BsonValue lte;
+
+    RangeSearchOperator(final List<SearchPath> path, @Nullable final String index,
+            @Nullable final BsonValue gt, @Nullable final BsonValue gte, @Nullable final BsonValue lt, @Nullable final BsonValue lte) {
+        super(index);
+        this.path = notNull("path", path);
+        this.gt = gt;
```

Comment:
We could do error checking here to ensure that only one of gt and gte and one of lt and lte are set.  Or we could just allow the server to error since it's a runtime error anyway.